app/moves/middleware.js

```diff
@@ -116,24 +115,46 @@ const movesMiddleware = {
           return moveService
             .getMovesCount({ dateRange, status: moveType.filter, locationId })
             .then(count => {
-              const { label } = moveType
               return {
-                label,
+                ...moveType,
                 value: count,
                 active: moveType.filter === req.params.status,
                 href: `${req.baseUrl}/${period}/${date}${
                   locationId ? '/' + locationId : ''
                 }/${moveType.filter}`,
               }
             })
-            .then(presenters.moveTypesToFilterComponent)
         })
       )
       next()
     } catch (error) {
       next(error)
     }
   },
+  setTotalMoves: (req, res, next) => {
+    const totalMoves = {
+      label: 'moves::dashboard.filter.total',
+      filter: 'total',
+      href: find(res.locals.moveTypeNavigation, { filter: 'proposed' }).href,
+    }
+    totalMoves.value = res.locals.moveTypeNavigation.reduce(
+      (accumulator, moveType) => {
+        return (accumulator += moveType.value)
+      },
+      0
+    )
+    res.locals.moveTypeNavigation = reject(res.locals.moveTypeNavigation, {
+      filter: 'proposed',
+    })
+    res.locals.moveTypeNavigation.unshift(totalMoves)
```

Comment:
Could use ES6 spread? I find it can be a little easier to read than `unshift`:

```javascript
res.locals.moveTypeNavigation = [
  ...totalMoves,
  reject(res.locals.moveTypeNavigation, {
    filter: 'proposed',
  }),
]
```

Reply:
Hmm... but really we want `unshift` there. I do agree that it's not an array method that it's used often, but it fits the case perfectly. I am not sure there's an argument to avoid it -- if anything, let's remind ourselves that it exists, too...

Reply:
It was more for readability. I think the addition of the spread operator in ES6 just makes what you're trying to do here more readable. But maybe that's more personal preference.

I think it might be more me thinking in a functional programming way to not mutate where possible and to return a new value instead.